tests/capi/unittest_capi_inference_nnfw_runtime.cc

```diff
@@ -15,108 +15,214 @@
 #include <ml-api-inference-internal.h>
 #include <ml-api-inference-pipeline-internal.h>
 
-static GMutex g_test_mutex;
-
-#define wait_for_sink(expected_cnt)                   \
-  do {                                                \
-    guint waiting_time = 0;                           \
-    gboolean done = FALSE;                            \
-    while (!done && waiting_time < 10000000U) {       \
-      g_mutex_lock (&g_test_mutex);                   \
-      done = (*sink_called_cnt >= expected_cnt);      \
-      waiting_time += 10000U;                         \
-      g_mutex_unlock (&g_test_mutex);                 \
-      if (!done)                                      \
-        g_usleep (10000U);                            \
-    }                                                 \
-    ASSERT_TRUE (done);                               \
-  } while (0)
+static GMutex g_test_mutex;   /** Global mutex lock for multiple sink callback */
```

Comment:
Q: should we implement global lock and count for test?
- wait_for_sink (guint * ptr_count, guint expected)
- gmutext in MLAPIInferenceNNFW, init and release on constructor/destructor

Reply:
After talking with you, I did a refactoring of this code without global mutex and count variable.
I think it looks good. Let me know your opinion.